outputs/kubernetes.go

```diff
@@ -63,35 +63,50 @@ func (k *KubernetesClient) Init() error {
 	return nil
 }
 
-func (k KubernetesClient) prepareInputs(input map[string]string) map[string]map[string]string {
-	a := make(map[string]map[string]string)
+func jsonOrString(input map[string]string, filter string) string {
+	var ret string
+	if ok := json.Valid([]byte(input["description"])); ok { // input is json
+		ret = gjson.Get(input["description"], filter).String()
+	} else {
+		ret = input["description"] // input is a string
+	}
+	return ret
+}
+
+func (k KubernetesClient) prepareInputs(input map[string]string) (string, map[string]map[string]string) {
+	retAction := make(map[string]map[string]string)
+	var retLabelSelector string
+
+	if strings.Contains(k.KubeLabelSelector, regoInputPrefix) {
+		retLabelSelector = jsonOrString(input, strings.TrimPrefix(k.KubeLabelSelector, regoInputPrefix+"."))
+	}
 
 	for key, m := range k.KubeActions {
 		for id, val := range m {
 			var calcVal string
 			if strings.HasPrefix(val, regoInputPrefix) {
-				if ok := json.Valid([]byte(input["description"])); ok { // input is json
-					calcVal = gjson.Get(input["description"], strings.TrimPrefix(val, regoInputPrefix+".")).String()
-				} else {
-					calcVal = input["description"] // input is a string
-				}
+				calcVal = jsonOrString(input, strings.TrimPrefix(val, regoInputPrefix+"."))
 			} else {
 				calcVal = val // no rego to parse
 			}
-			a[key] = map[string]string{id: calcVal}
+			if _, ok := retAction[key][id]; !ok && len(retAction[key]) <= 0 {
```

Comment:
when `len(retAction[key])` can be less than zero?

Reply:
Hopefully never, it is just a defensive programming carryover I have from languages where this can be possible.

Reply:
then probably better use `==` (to not make other developers raise similar questions)

Reply:
OK